### PR TITLE
[FIX] mail: fix inserting emoji at the correct position

### PR DESCRIPTION
**Pull request by Nemael**
This commit fixes emojis that were not inserted in the correct position in the composer

### CODE REVIEW

File: addons/mail/static/src/composer/composer.js [hunk 1]
```diff
@@ -79,7 +79,8 @@ export class Composer extends Component {
     }
 
     addEmoji(str) {
-        this.state.value += str;
+        let caret_position = this.ref.el.selectionStart;
+        this.state.value = this.state.value.substring(0, caret_position) + str + this.state.value.substring(caret_position, this.state.value.length);
```
Comment (alexkuhn) [inline addons/mail/static/src/composer/composer.js]: When adding emoji, it should concatenate as follow:
- before selection start
- replace between selection start-end with emoji
- after selection end

That way it properly handles any case of selection.

**Review by alexkuhn — CHANGES_REQUESTED** [review]

File: addons/mail/static/src/composer/composer.js [hunk 2]
```diff
@@ -20,6 +21,10 @@ export class Composer extends Component {
             (focus) => {
                 if (focus && this.ref.el) {
                     this.ref.el.focus();
+                    if(this.caretLocationBeforeAddingEmoji) {
+                        this.ref.el.setSelectionRange(this.caretLocationBeforeAddingEmoji, this.caretLocationBeforeAddingEmoji);
+                        this.caretLocationBeforeAddingEmoji = null;
+                    }
```
Comment (alexkuhn) [inline addons/mail/static/src/composer/composer.js]: Linter/prettier not happy with these LOCs,

Make sure to configure them properly on your machine. See `@web/tooling` (notably `enable.sh`)

Comment (alexkuhn) [inline addons/mail/static/src/composer/composer.js]: I was wondering why it was not just simply after `addEmoji` LOCs, but the current implementation of `useEffect` has side-effect to re-focus at the end of composer input...

`useEffect` is a bad smell equivalent to our `onChange` in the previous code of discuss. We should avoid it as much as possible.

I would suggest doing something to remove this `useEffect`. Make a test for coverage of chatter change of send message/log not in chatter, and reimplement a solution without this `useEffect`. If doing properly, then this "restore" of caret position should be applied immediately after `addEmoji`, not in a following `onChange`

**Comment by alexkuhn** [comment]
Again should rebase with target branch (commit history was changed)

**Comment by seb-odoo** [comment]
Do we still need this PR?

**Comment by Nemael** [comment]
> 

No, it was included in another PR, this one can be closed.

**Pull request closed without merging**